vyper/cli/vyper_compile.py

```diff
@@ -220,15 +239,35 @@ def _parse_args(argv):
         output_formats = ("archive_b64",)
 
     if args.no_optimize and args.optimize:
-        raise ValueError("Cannot use `--no-optimize` and `--optimize` at the same time!")
-
-    settings = Settings()
+        raise ValueError("Cannot use `--no-optimize` and `-O/--optimize` at the same time!")
 
-    # TODO: refactor to something like Settings.from_args()
+    optimize = None
     if args.no_optimize:
-        settings.optimize = OptimizationLevel.NONE
+        optimize = OptimizationLevel.NONE
     elif args.optimize is not None:
-        settings.optimize = OptimizationLevel.from_string(args.optimize)
+        # Handle both old-style (none, gas, codesize) and new-style (0, 1, 2, 3, s) arguments
+        opt_level = args.optimize.lower()
+        if opt_level in ["0", "1", "2", "3"]:
+            opt_level = "O" + opt_level
+        elif opt_level == "s":
+            opt_level = "Os"
+        optimize = OptimizationLevel.from_string(opt_level)
+
+    settings = Settings(optimize=optimize)
+
+    # Apply individual flag overrides
+    if args.no_inlining:
+        settings.venom_flags.disable_inlining = True
```

Comment:
can we do it like this?
```suggestion
    settings.venom_flags.disable_inlining  = args.no_inlining
```

Reply:
That would shadow the setting coming from venom_flags. I made it into |=